lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -0,0 +1,950 @@
+#include "TypeChecker.h"
+#include "swift/AST/Decl.h"
+#include "swift/AST/Expr.h"
+#include "swift/AST/Module.h"
+#include "swift/AST/ParameterList.h"
+#include "swift/AST/Pattern.h"
+#include "swift/AST/Stmt.h"
+#include "swift/AST/Types.h"
+#include "DerivedConformances.h"
+
+using namespace swift;
+using namespace DerivedConformance;
+
+// Returns whether the type represented by the given ClassDecl inherits from a
+// type which is Codable.
+static bool superclassIsCodable(TypeChecker &tc, ClassDecl *type) {
+  assert(type && "Cannot get superclass of null type.");
+  if (!type->hasSuperclass())
+    return false;
+
+  ASTContext &C = tc.Context;
+  auto parentDC = type->getDeclContext();
+  auto superType = type->getSuperclass();
+  return (bool)tc.conformsToProtocol(superType,
```

Comment:
I stand corrected; the return type is `Optional<swift::ProtocolConformanceRef>`, so the explicit cast is required.